velox/functions/lib/window/RowNumber.cpp

```diff
@@ -40,9 +41,17 @@ class RowNumberFunction : public exec::WindowFunction {
       vector_size_t resultOffset,
       const VectorPtr& result) override {
     int numRows = peerGroupStarts->size() / sizeof(vector_size_t);
-    auto* rawValues = result->asFlatVector<int64_t>()->mutableRawValues();
-    for (int i = 0; i < numRows; i++) {
-      rawValues[resultOffset + i] = rowNumber_++;
+    if (resultType_->isInteger()) {
+      auto* rawValues = result->asFlatVector<int32_t>()->mutableRawValues();
```

Comment:
Extract the below code as a templated function on resultType and just invoke it with the correct type in the apply function.

```
 auto* rawValues = result->asFlatVector<T>()->mutableRawValues();
 for (int i = 0; i < numRows; i++) {
      rawValues[resultOffset + i] = rowNumber_++;
  }
```

Reply:
Done.